src/strategies/garden-stakes/index.ts

```diff
@@ -0,0 +1,77 @@
+import { keccak256 } from '@ethersproject/keccak256';
+import { pack } from '@ethersproject/solidity';
+import { Multicaller, call } from '../../utils';
+import { BigNumber } from '@ethersproject/bignumber';
+
+export const author = 'gardenfi';
+export const version = '0.0.1';
+
+const abi = [
+  'function delegateNonce(address) external view returns (uint256)',
+  'function stakes(bytes32) external view returns (address owner, uint256 stake, uint256 units, uint256 votes, address filler, uint256 expiry)'
+];
+
+type Vote = {
+  votes: { _hex: string; _isBigNumber: boolean };
+};
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const multi = new Multicaller(network, provider, abi, { blockTag });
+  for (const address of addresses) {
+    const nonce = await getNonce(
+      provider,
+      address,
+      options.gardenStakerAddress
+    );
```

Comment:
Could we please move this out of the loop?

Reply:
Updated to use multi-call